pac4j-core/src/main/java/org/pac4j/core/client/Clients.java

```diff
@@ -104,7 +105,10 @@ protected void internalInit() {
      */
     public Client findClient(final WebContext context) {
         final String name = context.getRequestParameter(this.clientNameParameter);
-        CommonHelper.assertNotBlank("name", name);
+		if (name == null && defaultClient != null) {
```

Comment:
For a different need, we need a different pull request. Recommit the master file version.
Notice this must be discussed before submitting a pull request: do we really want to bring in that behaviour? I'm really not sure.